lib/vmdb/loggers/request_logger.rb

```diff
@@ -0,0 +1,72 @@
+module Vmdb::Loggers
+  class RequestLogger < ManageIQ::Loggers::Base
+    attr_reader :log_wrapper
+
+    def initialize(*args, log_wrapper:, **kwargs)
+      @log_wrapper = log_wrapper
+      super(*args, **kwargs)
+    end
+
+    private def add_to_db(severity, message = nil, progname = nil, resource_id: nil)
+      return [severity, message, progname] unless resource_id
+
+      # Adapted from Logger#add
+      severity ||= UNKNOWN
+      if severity < level
+        return [severity, message, progname]
+      end
+
+      if progname.nil?
+        progname = @progname
+      end
+      if message.nil?
+        if block_given?
+          message = yield
+        else
+          message = progname
+          progname = @progname
+        end
+      end
+
+      RequestLog.create(:message => message, :severity => format_severity(severity), :resource_id => resource_id) if resource_id
+
+      [severity, message, progname]
+    end
+
+    def info(progname = nil, resource_id: nil, &block)
```

Comment:
NOTE one thing I don't like about this approach is that it requires that the caller modify all calls to the logger, instead of passing it a logger instance that has everything already set up.

If instead we had e.g. a logger instance created with the object we're referencing then we can use normal `logger.info` calls.  We can also use standard wrapped loggers and pass the request logger into code that otherwise wouldn't have to know about it (aka Floe)

Reply:
Yeah, that's what we ended up having to do, but I don't think we can actually do the other way around as you're proposing.

Reply:
Hm that might be a bit of an issue for using this more generally then.  I was playing with overriding the automate subclass of this logger to log using this style maybe I go further down that path.

Reply:
I feel strongly about using a standard interface.

Would really prefer having the resource_id hardcoded in the logger (initializer) Like I did in https://github.com/ManageIQ/manageiq/pull/23406